pkg/apis/core/types.go

```diff
@@ -445,11 +445,11 @@ type PersistentVolumeClaimSpec struct {
 	// +optional
 	VolumeMode *PersistentVolumeMode
 	// This field can be used to specify either:
-	// * An existing VolumeSnapshot object (snapshot.storage.k8s.io/VolumeSnapshot - Beta)
+	// * An existing VolumeSnapshot object (snapshot.storage.k8s.io/VolumeSnapshot)
 	// * An existing PVC (PersistentVolumeClaim)
 	// * An existing custom resource/object that implements data population (Alpha)
-	// In order to use VolumeSnapshot object types, the appropriate feature gate
-	// must be enabled (VolumeSnapshotDataSource or AnyVolumeDataSource)
+	// In order to use custom resource/object types that implements data population,
+	// the AnyVolumeDataSource feature gate must be enabled.
 	// If the provisioner or an external controller can support the specified data source,
```

Comment:
L456 below: do we report an event if the data source is not supported? Can we remove that last sentence about future provisioner behavior changes?

Reply:
Removed the last sentence about future changes.  I also removed the sentence about reporting an event.  For AnyVolumeDataSource, the external-provisioner will let the external data populator to handle the volume provisioning so it will not report an event about the failure.  We are still working on the design of a data populator implementation so I'm not sure if we can put anything more specific here.